libs/core/src/components/grid/grid.ts

```diff
@@ -0,0 +1,126 @@
+import { LitElement, unsafeCSS } from 'lit'
+import { property } from 'lit/decorators.js'
+import {
+  gdsCustomElement,
+  html,
+} from '../../utils/helpers/custom-element-scoping'
+
+import { tokens } from '../../tokens.style'
+import grid from './grid.style.css'
+const gridCSS = unsafeCSS(grid)
+
+type GridSizes = 'none' | 'xs' | 's' | 'm' | 'l' | 'xl' | '2xl' | '3xl'
+type GridGap = GridSizes
+type GridPadding = GridSizes
+
+@gdsCustomElement('gds-grid')
+export class GdsGrid extends LitElement {
+  static styles = [tokens, gridCSS]
+
+  static shadowRootOptions: ShadowRootInit = {
+    mode: 'open',
+    delegatesFocus: true,
+  }
+
+  @property({ attribute: 'columns', type: String })
+  columns?: string
+
+  @property({ attribute: 'gap', type: String })
+  gap?: GridGap
+
+  @property({ attribute: 'padding', type: String })
+  padding?: GridPadding
+
+  @property({ attribute: 'fluid', type: String })
```

Comment:
Excellent, sounds good to me also, `auto-columns` makes sense. 
`columns="auto"` would be even more efficient but It might make it difficult to understand what is going on. 

Perhaps we could use same concept here if you want on `auto-columns` different sizes on different breakpoints. 
Will explore that option.

Reply:
Yea that's a great idea! 💎  We didn't even think about this

Reply:
This is updated now! 
`auto-columns` supports like other attributes multi values or single for all breakpoints.